src/main/kotlin/io/github/graphglue/connection/filter/definition/FilterDefinition.kt

```diff
@@ -42,51 +42,37 @@ class FilterDefinition<T : Node>(private val entryType: KClass<T>) :
      * @return the parsed [Filter]
      */
     fun parseFilter(value: Any?): Filter {
-        return Filter(parseMetaFilter(value))
-    }
-
-    /**
-     * Parses a meta filter.
-     * Requires that value contains exactly one of node, and, or, not are provided.
-     *
-     * @param value the meta filter to parse
-     * @return the parsed meta filter
-     * @throws IllegalStateException if the input type contains not one of node, and, or, not
-     */
-    private fun parseMetaFilter(value: Any?): MetaFilter {
-        value as Map<*, *>
-        if (value.size != 1) {
-            throw IllegalArgumentException("Exactly one of the fields [node, and, or, not] must be provided")
-        }
-        val (name, entry) = value.entries.first()
-        return when (name) {
-            "node" -> NodeMetaFilter(parseNodeFilter(entry!!))
-            "and" -> {
-                entry as List<*>
-                AndMetaFilter(entry.map(::parseMetaFilter))
-            }
-            "or" -> {
-                entry as List<*>
-                OrMetaFilter(entry.map(::parseMetaFilter))
-            }
-            "not" -> NotMetaFilter(parseMetaFilter(entry))
-            else -> throw IllegalStateException("Illegal input value which does not match GraphQL type")
-        }
+        return Filter(parseNodeFilter(value))
     }
 
     /**
      * Parses a node filter
-     * Requires that there is a definition present for all provided fields
+     * Requires for each field that either
+     * - a definition is present
+     * - it is an and/or/not meta filter field
      *
      * @param value the node filter to parse
      * @return the parsed filter
      */
-    private fun parseNodeFilter(value: Any): NodeFilter {
+    private fun parseNodeFilter(value: Any?): NodeFilter {
         value as Map<*, *>
         val entries = value.map {
             val (name, entry) = it
-            val definition = entries[name] ?: throw IllegalStateException("Unknown input")
-            definition.parseEntry(entry)
+            when (name) {
+                "and" -> {
+                    entry as List<*>
+                    AndMetaFilter(entry.map(::parseNodeFilter))
+                }
+                "or" -> {
+                    entry as List<*>
+                    OrMetaFilter(entry.map(::parseNodeFilter))
+                }
+                "not" -> NotMetaFilter(parseNodeFilter(entry))
```

Comment:
Allow __extensible__ meta filters, useful if somebody needs something like "more than 2/n true" (aka. or++ ?)

Reply:
We could do this by extending the mechanism used to provide additional filters, however this is not scope of this PR